diff --git a/arch/arm64/boot/dts/freescale/imx8mp-m8mp553.dts b/arch/arm64/boot/dts/freescale/imx8mp-m8mp553.dts
index 917fabc..c7bf1a3 100644
--- a/arch/arm64/boot/dts/freescale/imx8mp-m8mp553.dts
+++ b/arch/arm64/boot/dts/freescale/imx8mp-m8mp553.dts
@@ -8,6 +8,8 @@
 #include <dt-bindings/usb/pd.h>
 #include "imx8mp.dtsi"
 
+#define LVDS_PANEL_G215HVN010
+
 / {
 	model = "DFI M8MP553 Board";
 	compatible = "fsl,imx8mp-evk", "fsl,imx8mp";
@@ -239,6 +241,16 @@
 		status = "okay";
 	};
 
+	backlight_power: regulator-backlight_power {
+		compatible = "regulator-gpio";
+		regulator-name = "lcd_backlight_pwr";
+		regulator-min-microvolt = <0000001>;
+		regulator-max-microvolt = <3300000>;
+		gpios = <&gpio1 1 GPIO_ACTIVE_HIGH>;
+		gpios-states = <0>;
+		states = <3300000 0x1 0000001 0x0>;
+	};
+
 	panel-lvds0 {
 		compatible = "panel-lvds";
 		status = "okay";
@@ -248,8 +260,9 @@
 
 		backlight = <&lvds_backlight>;
 		power-supply = <&reg_lvds_panel>;
-		backlight-delay = <0>;
+		backlight-delay = <20>;
 
+		#if LVDS_PANEL_AM1024600DTZQWTD1H
 		panel-timing {
 			/* 1024x600 @60Hz : AM-1024600DTZQW-TD1H */
 			clock-frequency = <51000000>;
@@ -262,6 +275,22 @@
 			vfront-porch = <12>;
 			vback-porch = <23>;
 		};
+		#endif
+
+		#ifdef LVDS_PANEL_G215HVN010
+		panel-timing {
+			/* 1920x1080 @60Hz : G215HVN01.0 */
+			clock-frequency = <72000000>;
+			hactive = <1920>;
+			vactive = <1080>;
+			hsync-len = <44>;
+			hfront-porch = <88>;
+			hback-porch = <148>;
+			vsync-len = <5>;
+			vfront-porch = <4>;
+			vback-porch = <36>;
+		};
+		#endif
 
 		port {
 			panel_lvds0_in: endpoint {
@@ -275,7 +304,8 @@
 		pwms = <&pwm2 0 100000>; /* 10KHz*/
 		status = "okay";
 		pinctrl-0 = <&pinctrl_backlight_en>, <&pinctrl_backlight_pwm2>;
-		enable-gpios = <&gpio1 1 GPIO_ACTIVE_HIGH>;
+		power-supply = <&backlight_power>;
+		post-pwm-on-delay-ms = <130>;
 		brightness-levels = < 0  1  2  3  4  5  6  7  8  9
 				     10 11 12 13 14 15 16 17 18 19
 				     20 21 22 23 24 25 26 27 28 29
@@ -752,7 +782,8 @@
 
 &ldb {
 	status = "okay";
-	startup-delay-us = <20>;
+	fsl,dual-channel;
+	startup-delay-us = <40>;
 	lvds-channel@0 {
 		fsl,data-mapping = "spwg";
 		fsl,data-width = <24>;
diff --git a/drivers/gpu/drm/panel/panel-lvds.c b/drivers/gpu/drm/panel/panel-lvds.c
index 817114e..5d61316 100644
--- a/drivers/gpu/drm/panel/panel-lvds.c
+++ b/drivers/gpu/drm/panel/panel-lvds.c
@@ -103,13 +103,14 @@ static int panel_lvds_enable(struct drm_panel *panel)
 {
 	struct panel_lvds *lvds = to_panel_lvds(panel);
 
+	msleep(lvds->bl_delay);
 	if (lvds->backlight) {
 		lvds->backlight->props.state &= ~BL_CORE_FBBLANK;
 		lvds->backlight->props.power = FB_BLANK_UNBLANK;
 		backlight_update_status(lvds->backlight);
 	}
 
-	msleep(lvds->bl_delay);
+
 	return 0;
 }
 
diff --git a/drivers/video/backlight/pwm_bl.c b/drivers/video/backlight/pwm_bl.c
index 40ebb5f..50ae210 100644
--- a/drivers/video/backlight/pwm_bl.c
+++ b/drivers/video/backlight/pwm_bl.c
@@ -20,6 +20,9 @@
 #include <linux/regulator/consumer.h>
 #include <linux/slab.h>
 
+#define BL_VOLT_MAX 3300000
+#define BL_VOLT_MIN 0000001
+
 struct pwm_bl_data {
 	struct pwm_device	*pwm;
 	struct device		*dev;
@@ -60,6 +63,8 @@ static void pwm_backlight_power_on(struct pwm_bl_data *pb)
 	if (pb->post_pwm_on_delay)
 		msleep(pb->post_pwm_on_delay);
 
+	regulator_set_voltage(pb->power_supply, BL_VOLT_MAX, BL_VOLT_MAX);
+
 	if (pb->enable_gpio)
 		gpiod_set_value_cansleep(pb->enable_gpio, 1);
 
@@ -84,6 +89,7 @@ static void pwm_backlight_power_off(struct pwm_bl_data *pb)
 	state.duty_cycle = 0;
 	pwm_apply_state(pb->pwm, &state);
 
+	regulator_set_voltage(pb->power_supply, BL_VOLT_MIN, BL_VOLT_MIN);
 	regulator_disable(pb->power_supply);
 	pb->enabled = false;
 }
