From e8c050dba7bcb36deb77fb2a86c92fa3980685c4 Mon Sep 17 00:00:00 2001
From: "Slash.Huang" <slash.huang@dfi.com>
Date: Wed, 21 Apr 2021 10:56:05 +0800
Subject: [PATCH] disable CSI_SENSOR_KERNEL define to avoid reinstall same
 module

---
 vvcam/common/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b2b18cc..2d17e1d 100644
--- a/vvcam/common/Makefile
+++ b/vvcam/common/Makefile
@@ -31,7 +31,7 @@ EXTRA_CFLAGS += -I$(PWD)/../dwe/

 EXTRA_CFLAGS += -DISP8000NANO_V1802
 EXTRA_CFLAGS += -DWITH_DWE
-EXTRA_CFLAGS += -DCSI_SENSOR_KERNEL
+#EXTRA_CFLAGS += -DCSI_SENSOR_KERNEL

 EXTRA_CFLAGS += -DISP_REG_BASE0=0x32E10000
 EXTRA_CFLAGS += -DISP_REG_BASE1=0x32E20000
--
2.7.4

